\documentclass[12pt,conference,compsocconf]{../IEEEtran}
\usepackage{xltxtra}
%\usepackage{subfig}
%\usepackage{booktabs}
\usepackage{flushend}
\usepackage[numbers,sort&compress]{natbib}
\setmainfont{Times New Roman}

\begin{document}

\title{Chemical-Genetic Profiling and Its Applications}
\author
{
\IEEEauthorblockN
{
Hongjian Li
\IEEEauthorblockA
{
Department of Computer Science and Engineering\\
Chinese University of Hong Kong\\
hiji@cse.cuhk.edu.hk
}
}
}
\maketitle

\begin{abstract}

Chemical-genetic profiling of bioactive compounds has been repeatedly proved to be a powerful tool for identifying novel drugs, mode of actions of drugs and drug targets, and for better understanding the relation between genotype and phenotype. In the past two decades, research mainly concentrated on applying chemical-genetic profiling to the yeast genome, but seldom investigated into higher organisms like mammals, or higher level of practical applications like drug synergy. This survey briefly reviews the popular experimental techniques and computational methods for constructing and analyzing chemical-genetic profiles and discusses their pros and cons. In the foreseeable future, as both the number of bioactive compounds and the number of sequenced genes increase, more and more chemical-genetic profiles will emerge. How to establish an universal database to accommodate diverse output from various research groups, and how to correctly and properly utilize such ressources for better drug discovery will become challenging issues.

\end{abstract}

%\begin{IEEEkeywords}

%Chemical-Genetic Profiling, Drug-Target Pathway, Mode-of-Action

%\end{IEEEkeywords}

\section{Introduction}

Chemical genetics refers to the study of genetic responses to chemical compounds. In genome-wide level, it is called chemical genomics or chemogenomics. It embraces multiple early phase drug discovery technologies ranging from target identification and validation, through compound design and chemical synthesis, to biological testing and ADME profiling.

Chemical-genetic profiling is nowadays gaining more and more importance and popularity as it provides a valuable tool for a wide variety of biological applications, including but not limited to rapid identifications of novel drugs, mode of actions (MOAs) of drugs and drug targets, and understanding biological systems to a better extend by relating genotype and phenotype.

\section{Motivation}

Partly due to the massive failure of antimicrobial target-based \textit{in vitro} screening strategies over the past 20 years, chemical-genetic profiling is preferred over high-throughput screening in certain applications thanks to its ability to rapidly characterize compounds in an unbiased fashion and provide rich, functional information to elucidate their MOAs and targets.

High-throughput screening is mainly limited by two factors. The pharmaceutical targets are usually limited to a small subset of specific proteins, and it could be very difficult to conduct a comprehensive screening given the facts that the small molecule chemical space consists of as many as 10\textsuperscript{60} chemical compounds \citep{1104} and the rate of compound synthesis via combinational chemistry is constantly increasing. In contrast, chemical-genetic profiling can be genome-wide and even interspecies.

\section{Background}

Chemical-genetic profiling basically measures the quantitative values of growth defects of deletion strains under exposure to chemicals. The foundation of chemical-geneitic approach was laid with the idea to sequence the 12.5-megabase yeast genome in the 1980s. Figure \ref{fig:1081-1} shows the six major steps for yeast fitness profiling of pooled deletion strains. Experimental detail can be found in the original publicaton \citep{1081}.

\begin{figure}
\centering
\includegraphics[width=\linewidth]{1081-1.png}
\caption{Yeast fitness profiling of pooled deletion strains. Figure reprinted from \citep{1081}.}
\label{fig:1081-1}
\end{figure}

Figure \ref{fig:1082-1} compares genetic, chemical-genetic, and chemical-chemical profiling. In the case of chemical-genetic profiling, one gene product is inactivated via gene deletion and the other is inactivated by chemical inhibition (Figure \ref{fig:1082-1}a). Red indicates a significant drug-gene interaction (Figure \ref{fig:1082-1}b). Such quantative values can be used to predict the target process of unknown agents. In the case of chemical-genetic profiling, known compounds on the X-axis are clustered based on known mode of action or function, and the Y-axis shows clustering of other unknown genes with similar profiles.

\begin{figure}
\centering
\includegraphics[width=\linewidth]{1082-1.png}
\caption{Comparison of genetic, chemical-genetic, and chemical-chemical profiling. Figure reprinted from \citep{1082}.}
\label{fig:1082-1}
\end{figure}

With the quantative interaction values produced by chemical-genetic profiling, computational methods are then applied for in-depth analysis. Popular approaches include 2D hierarchical clustering \citep{1078,1079,1080}, probabilistic sparse matrix factorization (PSMF) \citep{1078} and bayesian factor model \citep{1079}. Such algorithmic approaches could provide potentially important insight into drug-target pathway \citep{1078,1079} or address the debate concerning the purpose of nonessential genes \citep{1080}.

\section{Compendium of Chemical-Genetic Interaction}

In \citeyear{1078}, \Citeauthor{1078} generated a compendium of chemical-genetic interaction profiles by testing the collection of viable yeast haploid deletion mutants for hypersensitivity to 82 compounds and natural product extracts using parallel fitness tests of large numbers of pooled deletion strains in a minimal amount of medium thanks to the unique molecular barcodes that tag and identify each deletion strain \citep{1078}. The 82 chemical perturbations contain 75 synthetic compounds and natural products, of which 23 are FDA approved, plus 7 crude antifungal extracts, derived from different marine sponges and microorganisms. The full list and the inhibitory concentrations can be found in the original publication \citep{1078}.

Having obtained the quantative gene-drug interaction values, they then applied both a hierarchical clustering and a PSMF method, which allows a gene or compound to be associated with more than one group. Eventually they managed to find tamoxifen, a breast cancer therapeutic, to disrupt calcium homeostasis, and find phosphatidylserine as a target for papuamide B, a cytotoxic lipopeptide with anti-HIV activity.

Figure \ref{fig:1078-1} shows the results of 2D clustering. 3,418 genes are plotted on the horizontal axis and 82 conditions are plotted on the vertical axis. Red lines indicate chemical-genetic interactions. They revealed that compounds with similar cellular effects showed similar chemical-genetic profiles and thereby cluster together on the vertical axis. They listed 11 clusters (i.e. clusters i to xi) for explanation. For instance, in cluster i, latrunculin B and cytochalasin A are both actin binding agents, and in cluster ii, both staurosporine and caspofungin are cell wall synthesis inhibitors. These examples demonstrate the power of 2D clustering for grouping compounds known to inhibit the same pathway or target. Likewise, on the horizontal axis, deletion mutants with similar chemical sensitivities also cluster together, grouping functionally related genes.

\begin{figure}
\centering
\includegraphics[width=\linewidth]{1078-1.png}
\caption{Two-dimensional hierarchical clustering analysis of chemical-genetic profiles. Figure reprinted from \citep{1078}.}
\label{fig:1078-1}
\end{figure}

One particular example is cluster ix, which groups tamoxifen and amiodarone together. Tamoxifen is a competitive inhibitor of estradiol binding to the estrogen receptor and a common breast cancer drug, while amiodarone is an antianginal and antiarrhythmic drug. This cluster suggested that tamoxifen may also produce antifungal activity, which was subsequently verified and confirmed by a series of biological assays \citep{1078}.

One major limitation of 2D clustering is its inability to associate a gene or compound with more than one group. To address this issue, the authors applied PSMF, allowing each cluster to be defined by an arbitrary set of genes and compounds (Figure \ref{fig:1078-2}). They identified 30 factors, which were referred to as subsets of compounds detected by the algorithm. They picked out factors 5, 6, 21, 29 for in-depth analysis. For example, in factor 5, papuamide B and alamethicin are the compounds that are most important in factor number 5, and this is driven by the common sensitivity of the deletion mutants listed on the y axis, with the most significant mutants at the top. Following spot dilution assays, they succeeded in linking Papuamide B, a high-molecular-weight cycliclipopeptide originally isolated from a marine sponge, to its cellular target, phosphatidylserine. For another example, in factor 29, extract 00-192, from a sea cucumber from the Commonwealth of Dominica and extract 00-132, derived from an Indonesian marine sponge, showed highly similar chemical-genetic profiles. From the two crude extracts, antifungal compounds were purified. The two compounds do not share structural features and thus chemical-genetic profiling appears to have linked molecules with disparate chemistry to the same biological activity.

\begin{figure}
\centering
\includegraphics[width=\linewidth]{1078-2.png}
\caption{Visualization of results from PSMF. Figure reprinted from \citep{1078}.}
\label{fig:1078-2}
\end{figure}


\section{Protein Complex-Based Bayesian Factor Model}

In \citeyear{1079}, \Citeauthor{1079} \citep{1079}.
\citep{1080}
\citep{1103} tutorial
\citep{1081} is state-of-the-art review.
\citep{1082} is state-of-the-art review.

Comparison
Have your own evaluation
Use others results and give reasons

\section{Discussion}

Summary on current works
  Avoid overlapping with existing reviews

\section{Prospectives}

PDB \citep{105} and ZINC \citep{532}
What will be done in the future?
  Consequences of current breakthroughs
  Bottleneck in the whole picture

\bibliographystyle{unsrtnat}
\bibliography{../refworks}

\end{document}
